agent.py

```diff
@@ -195,6 +195,12 @@ def run(self, config=None):
             if self.autorestart and self._should_restart():
                 self._do_restart()
 
+            # Check if we should run service discovery (this flag is set in the docker_daemon check).
```

Comment:
Can you combine that with those lines ? 
https://github.com/DataDog/dd-agent/blob/haissam/service-discovery/agent.py#L183-L184